java/arcs/core/host/ArcHostContextParticle.kt

```diff
@@ -195,15 +194,15 @@ class ArcHostContextParticle : AbstractArcHostParticle() {
      * Using instantiated particle to obtain [Schema] objects throught their
      * associated [EntitySpec], reconstruct an associated [Type] object.
      */
-    fun fromTag(arcId: String, particle: Particle, tag: String, handleName: String): Type {
+    fun fromTag(particle: Particle, tag: String, handleName: String): Type {
         try {
             val schema = particle.handles.getEntitySpec(handleName).SCHEMA
             return when (Tag.valueOf(tag)) {
                 Tag.Singleton -> SingletonType(EntityType(schema))
                 Tag.Collection -> CollectionType(EntityType(schema))
                 Tag.Entity -> EntityType(schema)
                 else -> throw IllegalArgumentException(
-                    "Illegal Tag $tag when deserializing ArcHostContext with ArcId $arcId"
+                    "Illegal Tag $tag when deserializing handle $handleName"
```

Comment:
I think it'll be good to keep the arcId here, because there are many handles named $handleName (recipes and particles typically reuse them) and so the log messages will be hard decipher unless you know what Arc this was for and can track down its recipe.

Reply:
If this exception is thrown, there's an outer exception handler that will wrap the exception with an exception message including the ArcId, which feels like the right way to me to include handling outer context.

Reply:
fair enough